docs/tutorials/on-device-training.md

```diff
@@ -0,0 +1,1406 @@
+
+For this tutorial, we will leverage the `MobileNetV2` model which has been trained on large-scale image datasets such as ImageNet (which has 1,000 classes). We will use this model for classifying custom data into one of four classes. The initial layers of MobileNetV2 serve as a feature extractor, capturing generic visual features applicable to various tasks, and only the final classifier layer will be trained for the task at hand.
+
+In this tutorial, we will use data to learn to:
+- Classify animals into one of four categories using a pre-packed animals dataset.
+- Classify celebrities into one of four categories using a custom celebrities dataset.
+
+## Prerequisites
+
+To follow this tutorial, you should have a basic understanding of Android app development using Java or Kotlin. Familiarity with C++ as well as familiarity with machine learning concepts such as neural networks and image classification will help as well.
+
+- Python development environment to prepare the training artifacts
+- Android Studio 4.1+
+- Android SDK 29+
+- Android NDK r21+
+- An Android device with a camera in [developer mode](https://developer.android.com/studio/debug/dev-options) with USB debugging enabled
+
+> **Note** The purpose of this tutorial is to show how ONNX Runtime can be used for On-Device Training. The author of this tutorial is not an expert in Android app development. The information provided in this tutorial is for educational purposes only and should not be considered as professional advice.
+
+> **Note** The entire android application is also made available on the [`onnxruntime-training-examples`](https://github.com/microsoft/onnxruntime-training-examples/tree/master/on_device_training/mobile/android/c-cpp) GitHub repository.
+
+## Offline Phase - Building the training artifacts
+
+1. Export the model to ONNX.
+
+    We start with a Pytorch model that has been pre-trained and export it to ONNX. The `MobileNetV2` model has been pretrained on the imagenet dataset that has data in 1000 categories. For our task of image classification, we want to only classify images in 4 classes. So, we change the last layer of the model to output 4 logits instead of 1,000.
+
+    ```python
+   import torch
+   import torchvision
+
+   model = torchvision.models.mobilenet_v2(
+      weights=torchvision.models.MobileNet_V2_Weights.IMAGENET1K_V2)
+
+   # The original model is trained on imagenet which has 1000 classes.
+   # For our image classification scenario, we need to classify among 4 categories.
+   # So we need to change the last layer of the model to have 4 outputs.
+   model.classifier[1] = torch.nn.Linear(1280, 4)
+
+   # Export the model to ONNX.
+   model_name = "mobilenetv2"
+   torch.onnx.export(model, torch.randn(1, 3, 224, 224),
```

Comment:
A link to documentation on exporting would be useful here. I find this is typically the most difficult step when integrating ONNX Runtime into any existing PyTorch pipeline.